src/types.rs

```diff
@@ -1555,15 +1861,52 @@ impl Rule {
         });
         lhs.chain(rhs).collect()
     }
-    /// Get a specific subterm in a Rule
+    /// Get a specific subterm in a `Rule`.
+    ///
+    /// See [`Term`] for more information.
+    ///
+    /// [`Term`]: enum.Term.html
+    ///
+    /// # Examples
+    /// 
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_term, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A(x_) = B | C(x_)").expect("parse of A(x_) = B | C(x_)");
+    ///
+    /// assert_eq!(r.at(&[0]).unwrap().display(&sig), "A(x_)");
+    /// assert_eq!(r.at(&[0,0]).unwrap().display(&sig), "x_");
+    /// assert_eq!(r.at(&[1,0]).unwrap().display(&sig), "B");
+    /// assert_eq!(r.at(&[1,1]).unwrap().display(&sig), "C(x_)");
+    /// ```
     pub fn at(&self, p: &[usize]) -> Option<&Term> {
         if p[0] == 0 {
             self.lhs.at(&p[1..].to_vec())
         } else {
             self.rhs[p[1]].at(&p[2..].to_vec())
         }
     }
-    /// Replace one subterm with another in a Rule
+    /// Replace one subterm with another in a `Rule`.
+    /// Returns a new `Rule` without changing the original.
+    ///
+    /// See [`Term`] for more information.
+    ///
+    /// [`Term`]: enum.Term.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, parse_term, parse_rule, Rule};
+    /// let mut sig = Signature::default();
+    ///
+    /// let mut r = parse_rule(&mut sig, "A(x_) = B | C(x_)").expect("parse of A(x_) = B| C(x_)");
+    /// let new_term = parse_term(&mut sig, "E").expect("parse of E");
+    /// let new_rule = r.replace(&[1,0], new_term);
```

Comment:
Agh! I'm pretty sure that this example, too, reflects a bug in the implementation. Could you please change the case that begins with `else if let Some(an_rhs) = self.rhs...` by subtracting one from each of the 4 indices there and then update the example accordingly? Sorry!